tracsubtickets/web_ui.py

```diff
@@ -117,14 +117,14 @@ def get_children(self, parent_id, db=None):
     def validate_ticket(self, req, ticket):
         action = req.args.get('action')
         if action == 'resolve':
-            db = self.env.get_db_cnx()
-            cursor = db.cursor()
-            cursor.execute("SELECT parent, child FROM subtickets WHERE parent=%s",
-                           (ticket.id, ))
-
-            for parent, child in cursor:
-                if Ticket(self.env, child)['status'] != 'closed':
-                    yield None, _('Child ticket #%s has not been closed yet') % child
+            with self.env.db_transaction as db:
+                cursor = db.cursor()
+                cursor.execute("SELECT parent, child FROM subtickets WHERE parent=%s",
+                               (ticket.id, ))
+
+                for parent, child in cursor:
+                    if Ticket(self.env, child)['status'] != 'closed':
+                        yield None, _('Child ticket #%s has not been closed yet') % child
```

Comment:
For proper i18n string extraction, this should be:
```
_("Child ticket #%(child)s has not been closed yet", child=child)
```

Use of double-quoted strings is consistent with first item in Trac coding guideline: https://trac.edgewall.org/wiki/TracDev/CodingStyle#Strings